File: GrayMattr.Samples/RestoreEmpty.fs
module RestoreEmpty

open GrayMattr

// Parse YAML front-matter

let str =
    @"---
---
This is content"

let file: GrayFile<obj> = Mattr.Parse str

let newFile: GrayFile<obj> =
    if file.IsEmpty then file.SetContent(str) else file

newFile |> ignore

File: GrayMattr.Samples.Csharp/RestoreEmpty.cs
namespace GrayMattr.Samples.Csharp.RestoreEmpty;

public class RestoreEmpty
{
    static void Main()
    {
        // Parse YAML front-matter

        string str = @"---
---
This is content";

        GrayFile<IDictionary<string, string>> file = Mattr.Parse<IDictionary<string, string>>(str);

        if (file.IsEmpty)
        {
            file = file.SetContent(str);
        }
    }
}

